lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1707,7 +1707,7 @@ class IOSTextEditingStrategy extends GloballyPositionedTextEditingStrategy {
     subscriptions.add(DomSubscription(activeDomElement, 'blur',
             (_) {
               final bool isFastCallback = blurWatch.elapsed < _blurFastCallbackInterval;
-              if (windowHasFocus && isFastCallback) {
+              if (windowHasFocus || isFastCallback) {
```

Comment:
Given the original introduction of `isFastCallback` here https://github.com/flutter/engine/pull/31718 , it is a bit strange that this code does not cause any test to fail.

From the explanation given here https://github.com/flutter/engine/pull/31718#issuecomment-1064975791 it sounds like we want to close the input connection when the "done" button is pressed on a keyboard. The logic without `isFastCallback` was only checking if the window had focus before refocusing the textfield. In the case of the "done" button this would cause the keyboard to stay open because the window has focus so the textfield is refocused. With `windowHasFocus && isFastCallback`, pressing the "done" button will close the input connection because it was not a fast callback. With `windowHasFocus || isFastCallback` it sounds like the "done" button would once again fall under the same issue where the keyboard can't hide because the window has focus. What do you think about this?

cc @mdebbar if you had any context on this issue.

Reply:
I didn't know this before. So we should find a way to detect the difference between clicking the `done` or `TapTextFieldTapRegion`.